### PR TITLE
Semigroupoid instances

### PR DESCRIPTION
**Pull request by alexfmpe**

### CODE REVIEW

File: src/Data/Map/Monoidal.hs [hunk 1]
```diff
@@ -216,6 +221,10 @@ instance (Ord k, Semigroup a) => Monoid (MonoidalMap k a) where
     {-# INLINE mappend #-}
 #endif
 
+instance Ord k => Alt (MonoidalMap k) where
```
Comment (alexfmpe) [inline src/Data/Map/Monoidal.hs]: GHC seems to get confused when trying to derive this one due to the `Applicative` constraint in `some/many`.

with `deriving Alt` I get 
```
src/Data/Map/Monoidal.hs:167:15-17: error:
    • Couldn't match type ‘M.Map k’ with ‘MonoidalMap k’
        arising from the coercion of the method ‘many’
          from type ‘forall a.
                     Applicative (M.Map k) =>
                     M.Map k a -> M.Map k [a]’
            to type ‘forall a.
                     Applicative (MonoidalMap k) =>
                     MonoidalMap k a -> MonoidalMap k [a]’
    • When deriving the instance for (Alt (MonoidalMap k))
    |
167 |               Alt, Apply, Bind)
```

while `deriving instance (Ord k) => Alt (MonoidalMap k)` gives me
```
src/Data/Map/Monoidal.hs:169:1-48: error:
    • Could not deduce (Applicative (M.Map k))
        arising from a use of ‘some’
      from the context: Ord k
        bound by the instance declaration
        at src/Data/Map/Monoidal.hs:169:1-48
      or from: Applicative (MonoidalMap k)
        bound by the type signature for:
```

File: src/Data/HashMap/Monoidal.hs [hunk 2]
```diff
@@ -42,6 +42,8 @@ import Prelude hiding (lookup, map)
 import Data.Maybe (fromMaybe)
 import Data.Semigroup
 import Data.Foldable (Foldable)
+import Data.Functor.Apply (Apply)
+import Data.Functor.Bind (Bind)
```
Comment (alexfmpe) [inline src/Data/HashMap/Monoidal.hs]: Only added these because there's no `Alt/Plus` for `HashMap`

**Comment by ali-abrar** [comment]
@alexfmpe Looks like we've got some conflicts because of some recent merges. Can you have a look?

File: src/Data/HashMap/Monoidal.hs (hunk 2 above)
Comment (alexfmpe) [reply src/Data/HashMap/Monoidal.hs]: There are now!

File: monoidal-containers.cabal [hunk 3]
```diff
@@ -54,12 +54,13 @@ library
                        hashable >= 1.2 && < 1.4,
                        lens >=4.4 && <5,
                        newtype >=0.2 && <0.3,
-                       semigroups >= 0.18 && < 0.20
+                       semigroups >= 0.18 && < 0.20,
+                       semigroupoids >= 5.0 && < 5.4
```
Comment (alexfmpe) [inline monoidal-containers.cabal]: I don't recall why I picked `>= 5.0`. `lens 4.4` requires `semigroupoids ==4.*` so maybe `>= 4.0` is better?
AFAICT Map/IntMap instances were always present, and I added CPP for the HashMap ones .

**Comment by alexfmpe** [comment]
@ali-abrar @bgamari PR is conflict free again, but CI seems blocked on
_2 workflows awaiting approval_
_First-time contributors need a maintainer to approve running workflows._

**Comment by ali-abrar** [comment]
Approved the workflow

**Comment by alexfmpe** [comment]
CI is green

**Review by ali-abrar — APPROVED** [review]

**Comment by alexfmpe** [comment]
It's been 5 years. I have no interest in more rounds of fixing conflicts.

**Pull request closed without merging**